clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -140,38 +207,92 @@ public Struct parseResponse(short version, ByteBuffer buffer) {
             return parseResponse(version, buffer, (short) 0);
         }
     },
-    CREATE_TOPICS(19, "CreateTopics", CreateTopicsRequest.schemaVersions(), CreateTopicsResponse.schemaVersions()),
-    DELETE_TOPICS(20, "DeleteTopics", DeleteTopicsRequest.schemaVersions(), DeleteTopicsResponse.schemaVersions()),
-    DELETE_RECORDS(21, "DeleteRecords", DeleteRecordsRequest.schemaVersions(), DeleteRecordsResponse.schemaVersions()),
-    INIT_PRODUCER_ID(22, "InitProducerId", InitProducerIdRequest.schemaVersions(),
-            InitProducerIdResponse.schemaVersions()),
-    OFFSET_FOR_LEADER_EPOCH(23, "OffsetForLeaderEpoch", true, OffsetsForLeaderEpochRequest.schemaVersions(),
-            OffsetsForLeaderEpochResponse.schemaVersions()),
+    CREATE_TOPICS(19, "CreateTopics",
+            "Sent by an (admin) client to the controller to create the topics given in the request.",
+            CreateTopicsRequest.schemaVersions(), CreateTopicsResponse.schemaVersions()),
+    DELETE_TOPICS(20, "DeleteTopics",
+            "Sent by an (admin) client to the controller to delete the topics given in the request.",
+            DeleteTopicsRequest.schemaVersions(), DeleteTopicsResponse.schemaVersions()),
+    DELETE_RECORDS(21, "DeleteRecords",
+            "Sent by an client to the controller to request the deletion of records from topics, " +
```

Comment:
It's sent by the (admin) client to the leader for the topics partitions included in the request